File: python/examples/accounts/homepages/v1/unclaim_homepage_sample.py
# -*- coding: utf-8 -*-
# Copyright 2025 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""A module to unclaim a Homepage."""

# [START merchantapi_unclaim_homepage]
from examples.authentication import configuration
from examples.authentication import generate_user_credentials
from google.shopping.merchant_accounts_v1beta import HomepageServiceClient
from google.shopping.merchant_accounts_v1beta import UnclaimHomepageRequest

_ACCOUNT = configuration.Configuration().read_merchant_info()


def unclaim_homepage():
  """Unclaims the homepage for a given Merchant Center account."""

  # Gets OAuth Credentials.
  credentials = generate_user_credentials.main()

  # Creates a client.
  client = HomepageServiceClient(credentials=credentials)

  # Creates Homepage name to identify Homepage.
  # The name has the format: accounts/{account}/homepage
  name = "accounts/" + _ACCOUNT + "/homepage"

  # Creates the request.
  request = UnclaimHomepageRequest(name=name)

  # Makes the request and catches and prints any error messages.
  try:
    response = client.unclaim_homepage(request=request)
    print(f"Unclaimed Homepage: {response}")
  except RuntimeError as e:
    print(f"Failed to unclaim homepage: {e}")


if __name__ == "__main__":
  unclaim_homepage()

# [END merchantapi_unclaim_homepage]

File: php/examples/accounts/homepages/v1/UnclaimHomepageSample.php
<?php
/**
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
require_once __DIR__ . '/../../../../vendor/autoload.php';
require_once __DIR__ . '/../../../Authentication/Authentication.php';
require_once __DIR__ . '/../../../Authentication/Config.php';
// [START merchantapi_unclaim_homepage]
use Google\ApiCore\ApiException;
use Google\Shopping\Merchant\Accounts\V1\Client\HomepageServiceClient;
use Google\Shopping\Merchant\Accounts\V1\UnclaimHomepageRequest;

/**
 * This class demonstrates how to unclaim the homepage for a given Merchant Center account.
 */
class UnclaimHomepage
{
    /**
     * Unclaims the homepage for a given Merchant Center account.
     *
     * @param array $config The configuration data for authentication and account ID.
     * @return void
     * @throws ApiException if the API call fails.
     */
    public static function unclaimHomepageSample(array $config): void
    {
        // Gets the OAuth credentials to make the request.
        $credentials = Authentication::useServiceAccountOrTokenFile();

        // Creates options config containing credentials for the client to use.
        $options = ['credentials' => $credentials];

        // Creates a client.
        $homepageServiceClient = new HomepageServiceClient($options);

        // Creates Homepage name to identify Homepage.
        // The name has the format: accounts/{account}/homepage
        $name = "accounts/" . $config['accountId'] . "/homepage";

        // Calls the API and catches and prints any network failures/errors.
        try {
            $request = new UnclaimHomepageRequest(['name' => $name]);

            print "Sending Unclaim Homepage request:\n";

            $response = $homepageServiceClient->unclaimHomepage($request);

            print "Retrieved Homepage below\n";
            print_r($response);
        } catch (ApiException $e) {
            print $e->getMessage();
        }
    }

    /**
     * Helper to execute the sample.
     *
     * @return void
     */
    public function callSample(): void
    {
        $config = Config::generateConfig();

        // Makes the call to unclaim the homepage.
        self::unclaimHomepageSample($config);
    }
}


// Run the script
$sample = new UnclaimHomepage();
$sample->callSample();
// [END merchantapi_unclaim_homepage]
